app/src/main/java/com/pavesid/androidacademy/ui/movies/MoviesViewModel.kt

```diff
@@ -14,28 +15,49 @@ import kotlinx.coroutines.CoroutineExceptionHandler
 import kotlinx.coroutines.launch
 import timber.log.Timber
 
-class MoviesViewModel @ViewModelInject constructor(
+internal class MoviesViewModel @ViewModelInject constructor(
     private val repository: MoviesRepository,
     @IODispatcher private val dispatcher: CoroutineDispatcher
 ) : ViewModel() {
 
     private val _movies = MutableLiveData<Resource<List<Movie>>>()
     val movies: LiveData<Resource<List<Movie>>> = _movies
 
-    private var list = emptyList<Movie>()
+    private var page = 1
+    private var isLoading = false
+
+    private var list = mutableListOf<Movie>()
 
     private val exceptionHandler = CoroutineExceptionHandler { _, throwable ->
         _movies.postValue(Resource.error(throwable.message.orEmpty(), null))
         Timber.d(throwable)
     }
 
     init {
-        viewModelScope.launch(dispatcher + exceptionHandler) {
-            _movies.postValue(Resource.loading(null))
-            list = repository.getMovies()
-            _movies.postValue(
-                Resource.success(list)
-            )
+        loadMovies()
+    }
+
+    /**
+     * Called when a new data packet needs to be received
+     */
+    @MainThread
+    fun loadMovies() {
+        if (!isLoading) {
+            viewModelScope.launch(dispatcher + exceptionHandler) {
+                isLoading = true
+                if (list.isEmpty()) {
+                    _movies.postValue(Resource.loading(null))
+                }
+                val movies = repository.getMovies(page)
+                if (movies.isNotEmpty()) {
+                    list.addAll(movies)
+                    page++
+                    _movies.postValue(
+                        Resource.success(list)
+                    )
+                }
+                isLoading = false
+            }
```

Comment:
You need to check if the last page loaded. Try to scroll the list to the last page :)